pkg/services/authn/clients/ext_jwt.go

```diff
@@ -99,87 +97,91 @@ func (s *ExtendedJWT) IsEnabled() bool {
 }
 
 func (s *ExtendedJWT) authenticateAsUser(
-	idTokenClaims claims.IdentityClaims,
-	accessTokenClaims claims.AccessClaims,
+	idTokenClaims *authlib.Claims[authlib.IDTokenClaims],
+	accessTokenClaims *authlib.Claims[authlib.AccessTokenClaims],
 ) (*authn.Identity, error) {
 	// Only allow id tokens signed for namespace configured for this instance.
-	if allowedNamespace := s.namespaceMapper(s.getDefaultOrgID()); !claims.NamespaceMatches(idTokenClaims, allowedNamespace) {
-		return nil, errExtJWTDisallowedNamespaceClaim.Errorf("unexpected id token namespace: %s", idTokenClaims.Namespace())
+	if allowedNamespace := s.namespaceMapper(s.getDefaultOrgID()); !claims.NamespaceMatches(authlib.NewIdentityClaims(*idTokenClaims), allowedNamespace) {
+		return nil, errExtJWTDisallowedNamespaceClaim.Errorf("unexpected id token namespace: %s", idTokenClaims.Rest.Namespace)
```

Comment:
nil pointer dereference potential here?

Reply:
nope, because the caller to the function ensures that doesn't happen.

https://github.com/grafana/grafana/blob/76683b17c3db266099b0d6ffc0c0b53305cd3f9a/pkg/services/authn/clients/ext_jwt.go#L84-L89

Reply:
I could remove the pointers in the function signature, maybe better that way

Reply:
90b246de6ef9d119538633b37a761c3cc6182be7

Reply:
much better